### PR TITLE
BaSP-MR-35: TRAINERS TEST- get, getById and create

### PR DESCRIPTION
**Pull request by ticilicarzze**

### CODE REVIEW

File: src/tests/trainers.test.js [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+});
+
+describe('GET /api/trainers', () => {
+  test('Should return status 200', async () => {
+    const response = await request(app).get('/api/trainers').send();
+    expect(response.status).toBe(200);
+  });
+
+  test('Should return status 404', async () => {
+    const response = await request(app).get('/api/trainersadasd').send();
```
Comment (JuanManuel-Blois) [inline src/tests/trainers.test.js:28]: This test is going to pass as a false positive but is not reflective of the actual mistake, try console login the response to see the error message to double check

File: src/tests/trainers.test.js [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+});
+
+describe('GET /api/trainers/:id', () => {
+  test('Should return status 200', async () => {
+    const response = await request(app).get('/api/trainers/64693b1d0b3782bf437c3826').send();
+    expect(response.status).toBe(200);
+  });
+
+  test('Should return status 404', async () => {
+    const response = await request(app).get('/api/trainer/64693b1d0b3782bf437c3821').send();
```
Comment (JuanManuel-Blois) [inline src/tests/trainers.test.js:40]: This test is going to pass as a false positive but is not reflective of the actual mistake, try console login the response to see the error message to double check

File: src/tests/trainers.test.js [hunk 3]
```diff
@@ -1,5 +1,55 @@
...
+});
+
+describe('POST /api/trainers/', () => {
+  test('Should return status 201', async () => {
+    const response = await request(app).post('/api/trainers').send(mockTrainer);
+    expect(response.status).toBe(201);
+  });
+
+  test('Should return status 404', async () => {
+    const response = await request(app).post('/api/trainer').send();
```
Comment (JuanManuel-Blois) [inline src/tests/trainers.test.js:52]: This test is going to pass as a false positive but is not reflective of the actual mistake, try console login the response to see the error message to double check

**Comment by JuanManuel-Blois** [comment]
You need to restore the deleted files to avoid conflicts with future PR's.

File: src/tests/trainers.test.js [hunk 4]
```diff
@@ -1,5 +1,55 @@
...
+};
+
+beforeAll(async () => {
+  await Trainer.collection.insertMany(trainerSeed);
+});
+
+describe('GET /api/trainers', () => {
+  test('Should return status 200', async () => {
+    const response = await request(app).get('/api/trainers').send();
+    expect(response.status).toBe(200);
```
Comment (nicolasblasco) [inline src/tests/trainers.test.js:24]: Don't you think we can test the body of the response? We can test the message, data, and error of the body of the response.

We should do this for all of our tests.

File: src/tests/trainers.test.js [hunk 5]
```diff
@@ -1,5 +1,55 @@
...
+
+  test('Should return status 404', async () => {
+    const response = await request(app).get('/api/trainersadasd').send();
+    expect(response.status).toBe(404);
+  });
+});
+
+describe('GET /api/trainers/:id', () => {
+  test('Should return status 200', async () => {
+    const response = await request(app).get('/api/trainers/64693b1d0b3782bf437c3826').send();
```
Comment (nicolasblasco) [inline src/tests/trainers.test.js:35]: What happens if we change the trainer ID in the seed file? I think we should capture the id of the first trainer from the seed into a variable. Then we can use it for our tests.

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
We should add a description to our pull request

**Review by LucianoAlarcon — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**